dockstore-webservice/src/main/java/io/dockstore/webservice/core/ChecksumConverter.java

```diff
@@ -0,0 +1,50 @@
+package io.dockstore.webservice.core;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.persistence.AttributeConverter;
+import javax.persistence.Converter;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Converter(autoApply = true)
+public class ChecksumConverter implements AttributeConverter<List<Checksum>, String> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ChecksumConverter.class);
+    @Override
+    public String convertToDatabaseColumn(List<Checksum> checksums) {
+        String stringChecksums;
+        if (checksums != null && !checksums.isEmpty()) {
+            try {
+                stringChecksums = checksums.stream().map(checksum -> checksum.getType().trim() + ":" + checksum.getChecksum().trim()).collect(Collectors.joining(","));
+            } catch (NullPointerException ex) {
+                LOG.error("Could not convert checksum(s) to string for database", ex);
+                return null;
```

Comment:
I'm not sure if this is right or not. Let's say somehow a null value got passed in -- although you're logging the error, it would essentially silently fail -- a null gets written out to the database, whereas the caller might have expected checksums to have been written out.

But I don't know the Hibernate pattern for this sort of thing

Reply:
Alternatively, this could be a database constraint. 
This column should pick two out of (content, null, empty content) and keep it consistent.

Reply:
@denis-yuen not sure if that applies since the checksums are being stored as a string in the db

Reply:
I think it could, but it could be a follow-up ticket. 
You can setup constraints with things like https://docs.oracle.com/javaee/6/api/javax/persistence/Column.html#nullable%28%29 or https://stackoverflow.com/questions/7951120/sql-not-empty-instead-of-not-null

Basically it simplifies things if you don't need to write code `(if !null && !value.isEmpty())` 
You also don't need to worry about the failure that Charles is proposing above.

Reply:
The case I am concerned about is starting from Java, before the database constraints come into play, at least as I understand it. Something like this: 

```
Checksum checkSum = new CheckSum();
checksum.setType(null); // Already null, but just to emphasize the point
checksum.setChecksum("1234");
Image image = new Image();
image.setChecksums(Collections.singleton(checksum));
// The above will cause the NPE in this method when being converted.
```